flake.nix

```diff
@@ -0,0 +1,38 @@
+{
+  description = "iKnow's shared nix utilities";
+
+  inputs.flake-utils.url = "github:numtide/flake-utils";
+  inputs.nixpkgs.url = "github:nixos/nixpkgs/nixpkgs-unstable";
+  inputs.branchctl.url = "git+ssh://git@github.com/iknow/branchctl?ref=flakes";
+  inputs.branchctl.inputs.nixpkgs.follows = "nixpkgs";
+  inputs.branchctl.inputs.flake-utils.follows = "flake-utils";
+
+  outputs = { self, nixpkgs, flake-utils, branchctl }:
+    let
+      supportedSystems = [ "x86_64-linux" "x86_64-darwin" "aarch64-linux" "aarch64-darwin" ];
+      overlay = import ./overlays { inherit branchctl; };
```

Comment:
I'd like to drop the duplicated branchctl packages and use
```nix
{
  overlay = nixpkgs.lib.composeExtensions branchctl.overlay (import ./overlays);
}
```

But I don't see a nice way to do this while maintaining support for non-flake systems.